adam/perception/__init__.py

```diff
@@ -4,10 +4,10 @@
 """
 from abc import ABC, abstractmethod
 from enum import Enum
-from typing import Generic, Optional, Tuple, TypeVar
+from typing import Generic, Optional, Tuple, TypeVar, Mapping, Any
 
 from attr import attrib, attrs
-from attr.validators import instance_of, optional
+from attr.validators import instance_of, optional, deep_mapping
```

Comment:
Was `deep_mapping` needed for something else besides action features? It looks like an unused import.

Reply:
Huh this is, pylint didn't catch it but pycharm did!

Reply:
Strange!